### PR TITLE
ci: use starflow policy and QA workflows

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `make lint && make test`?

-----

### CODE REVIEW

File: .github/workflows/qa.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      - "release/*"
+      - "renovate/*"
+  pull_request:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: lengau/starflow/lint@work/CRAFT-3602/test-action
```
Comment (lengau) [inline .github/workflows/qa.yaml:17]:
```suggestion
      - uses: canonical/starflow/lint@main
```

File: .github/workflows/qa.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          - [noble, arm64]  # Self-hosted Ubuntu Noble runner to test arm64
+          - windows-latest
+          - macos-15  # arm64
+          - macos-13  # amd64
+    runs-on: ${{ matrix.platform }}
+    env:
+      UV_PYTHON: ${{ matrix.python }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: lengau/starflow/test-python@work/CRAFT-3602/test-action
```
Comment (lengau) [inline .github/workflows/qa.yaml:33]:
```suggestion
      - uses: canonical/starflow/test-python@main
```

File: .github/workflows/qa.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          - ubuntu-latest
+          - windows-latest
+          - macos-15  # arm64
+          - macos-13  # amd64
+    runs-on: ${{ matrix.platform }}
+    env:
+      UV_PYTHON: ${{ matrix.python }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: lengau/starflow/test-python@work/CRAFT-3602/test-action
```
Comment (lengau) [inline .github/workflows/qa.yaml:49]:
```suggestion
      - uses: canonical/starflow/test-python@main
```

File: .github/workflows/policy.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+name: Check policy
+on:
+  pull_request:
+
+jobs:
+  policy:
+    uses: lengau/starflow/.github/workflows/policy.yaml@work/CRAFT-3602/test-workflows
```
Comment (lengau) [inline .github/workflows/policy.yaml:7]:
```suggestion
    uses: canonical/starflow/.github/workflows/policy.yaml@main
```

File: .github/workflows/qa.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+      - "main"
+      - "feature/*"
+      - "hotfix/*"
+      - "release/*"
+      - "renovate/*"
+  pull_request:
+
+jobs:
+  lint:
+    uses: lengau/starflow/.github/workflows/lint-python.yaml@work/CRAFT-3602/test-workflows
```
Comment (lengau) [inline .github/workflows/qa.yaml:14]:
```suggestion
    uses: canonical/starflow/.github/workflows/lint-python.yaml@main
```

File: .github/workflows/qa.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  test:
+    uses: lengau/starflow/.github/workflows/test-python.yaml@work/CRAFT-3602/test-workflows
```
Comment (lengau) [inline .github/workflows/qa.yaml:16]:
```suggestion
    uses: canonical/starflow/.github/workflows/test-python.yaml@main
```

**Review by mattculler — APPROVED** [review]

File: Makefile [hunk 7]
```diff
@@ -6,95 +6,211 @@ ifneq ($(OS),Windows_NT)
...
+setup: install-uv setup-precommit ## Set up a development environment
+	uv sync --frozen --all-extras
+
+.PHONY: setup-tests
+setup-tests: install-uv install-venv  ##- Set up a testing environment without linters
+	uv sync --frozen
+
+.PHONY: setup-lint
+setup-lint: install-uv install-shellcheck install-ruff install-codespell ##- Set up a linting environment
+	uv sync --frozen --no-dev --no-install-workspace --extra lint --extra types
```
Comment (dariuszd21) [inline Makefile:47]: Should we also do `--no-install-workspace` for `setup` and `setup-tests` and ?

File: Makefile [hunk 8]
```diff
@@ -6,95 +6,211 @@ ifneq ($(OS),Windows_NT)
...
+	success=true
+	ruff check --fix $(SOURCES) || success=false
+	ruff format $(SOURCES)
+	$$success || exit 1
 
-.PHONY: test
-test: test-unit test-integration  ## Run all tests with the default python
+.PHONY: format-codespell
+format-codespell:  ##- Fix spelling issues with codespell
+	uv run codespell --toml pyproject.toml --write-changes $(SOURCES)
```
Comment (dariuszd21) [inline Makefile:74]: Will it use the default `codespell` (installed as `snap` package above) ?

File: pyproject.toml [hunk 9]
```diff
@@ -110,10 +110,12 @@ line_length = 88
 minversion = "7.0"
 testpaths = "tests"
 xfail_strict = true
+markers = [
+    "slow: marks tests as slow (deselect with '-m \"not slow\"')",
+]
```
Comment (dariuszd21) [inline pyproject.toml:115]: I couldn't spot using this marker in `Makefile`. Is it intended ?

File: Makefile [hunk 10]
```diff
@@ -6,95 +6,211 @@ ifneq ($(OS),Windows_NT)
...
+else ifeq ($(OS),Windows_NT)
+	powershell -c "irm https://astral.sh/uv/install.ps1 | iex"
+else
+	curl -LsSf https://astral.sh/uv/install.sh | sh
+endif
+
+.PHONY: install-venv
+install-venv:
+ifneq ($(shell which apt-get),)
+	python3 -m venv --help && sudo apt-get --yes install python3-venv
```
Comment (dariuszd21) [inline Makefile:186]: Is it a good operator ? It seems to me as it should be `or`
```suggestion
	python3 -m venv --help || sudo apt-get --yes install python3-venv
```

Btw. is there still code that requires a normal "venv" ?

**Review by dariuszd21 — APPROVED** [review]
Looks good to me. 
I've left few notes there to better understand few things :sweat_smile:

File: Makefile (hunk 7 above)
Comment (lengau) [reply Makefile:47]: No, because for those we actually want to install the package and its dependencies.

File: Makefile (hunk 8 above)
Comment (lengau) [reply Makefile:74]: Good catch!
```suggestion
	codespell --toml pyproject.toml --write-changes $(SOURCES)
```

File: Makefile (hunk 10 above)
Comment (lengau) [reply Makefile:186]: The starbase `test_setuptools.py` test does.

File: pyproject.toml (hunk 9 above)
Comment (lengau) [reply pyproject.toml:115]: Yes, it's used as a `PYTEST_ADDOPTS` item for the [slow tests](https://github.com/canonical/craft-platforms/actions/runs/12058519983/job/33625373752?pr=60#step:6:5)

**Pull request merged**